pkg/workloads/cortex/lib/client/onnx.py

```diff
@@ -12,79 +12,500 @@
 
         Returns:
-            numpy.ndarray: The prediction returned from the model.
+            The prediction returned from the model.
+
+        Raises:
+            UserRuntimeException if the validation fails.
         """
-        if consts.SINGLE_MODEL_NAME in self._model_names:
-            return self._run_inference(model_input, consts.SINGLE_MODEL_NAME)
 
-        if model_name is None:
-            raise UserRuntimeException(
-                "model_name was not specified, choose one of the following: {}".format(
-                    self._model_names
+        model_name, model_version = self._validate_model_args(model_name, model_version)
+        return self._run_inference(model_input, model_name, model_version)
+
+    def _run_inference(self, model_input: Any, model_name: str, model_version: str) -> Any:
+        """
+        Run the inference on model model_name of version model_version.
+        """
+
+        try:
+            model = self._get_model(model_name, model_version)
+            if model is None:
+                raise UserRuntimeException(
+                    f"model {model_name} of version {model_version} wasn't found"
                 )
+        except Exception as e:
+            raise
+
+        try:
+            input_dict = convert_to_onnx_input(model_input, model["signatures"], model_name)
+            return model["session"].run([], input_dict)
+        except Exception as e:
+            raise UserRuntimeException(
+                f"failed inference with model {model_name} of version {model_version}", str(e)
             )
 
-        if model_name not in self._model_names:
+    def get_model(self, model_name: Optional[str] = None, model_version: str = "highest") -> dict:
+        """
+        Validate input and then return the model loaded into a dictionary.
+        The counting of tag calls is recorded with this method (just like with the predict method).
+
+        Args:
+            model_name: Model to use when multiple models are deployed in a single API.
+            model_version: Model version to use. Can also be "highest" for picking the highest version or "latest" for picking the most recent version.
+
+        Returns:
+            The model as returned by _load_model method.
+
+        Raises:
+            UserRuntimeException if the validation fails.
+        """
+        model_name, model_version = self._validate_model_args(model_name, model_version)
+        try:
+            model = self._get_model(model_name, model_version)
+            if model is None:
+                raise UserRuntimeException(
+                    f"model {model_name} of version {model_version} wasn't found"
+                )
+        except Exception as e:
+            raise
+        return model
+
+    def _get_model(self, model_name: str, model_version: str) -> Any:
```

Comment:
This function seems to almost be identical to client.python._get_model (same comments apply here)